plugins/woocommerce/client/legacy/js/frontend/checkout.js

```diff
@@ -550,7 +556,14 @@ jQuery( function( $ ) {
 
 							// Add new errors
 							if ( result.messages ) {
-								wc_checkout_form.submit_error( result.messages );
+								var $msgs = $( result.messages )
+									.removeAttr( 'role' )
+									.attr( 'tabindex', '-1' );
+								var $msgsWithLink = wc_checkout_form.wrapMessagesInsideLink( $msgs );
+								var $msgsWrapper = $( '<div role="alert"></div>' ).append( $msgsWithLink );
+
+								wc_checkout_form.submit_error( $msgsWrapper.prop( 'outerHTML' ) );
```

Comment:
The `submit_error` method expects the message content as a string. The conversion is done by calling `prop( 'outerHTML' )`.